Fix copyright - Granola-Team/mina-indexer#516

jhult opened the pull request.

Isaac-DeFrain reviewed (approved):
LGTM I'll approve but I'd prefer for someone else to pull the trigger. This is outside my 🌟zone-of-genius🌟

jhult commented:
I did keep the original copyright in the `NOTICE` files (example [here](https://github.com/Granola-Team/mina-indexer/blob/baac686580257356cb8043b3e4d19e0f26f85591/mina_serialization_macros/NOTICE)). Since the [`mina-rs` license](https://github.com/ChainSafe/mina-rs/blob/main/LICENSE) was Apache 2.0, I formed my `NOTICE` based on the Apache example [here](https://infra.apache.org/licensing-howto.html#example-notice).

trevorbernard commented on mina_serialization_macros/proc_macros/Cargo.toml:
```diff
@@ -1,5 +1,5 @@
 [package]
-authors = ["Granola <hello@granola.team>"]
+authors = ["Mina Foundation, Inc."]
```
I think we should retain the original authors value. @robinbb ?

jhult replied:
Updated.

`authors` now reads:

> ChainSafe Systems Inc., Granola Systems Inc.

trevorbernard commented on mina_serialization_macros/versioned/Cargo.toml:
```diff
@@ -1,5 +1,5 @@
 [package]
-authors = ["Granola <hello@granola.team>"]
+authors = ["Mina Foundation, Inc."]
```
See previous comment

jhult replied:
Updated.

jhult commented:
Merge #523 first. Then rebase on `main`.

jhult commented:
@robinbb, requesting review.

I rebased (and forced pushed) after the [merge of the Apache 2.0 license](https://github.com/Granola-Team/mina-indexer/commit/f838955feb36240c87bd323e5cd25fc0409ed419). Since the entire project is now Apache 2.0 licensed, I removed the additional `LICENSE` files.
